src/lv_core/lv_obj.c

```diff
@@ -491,6 +496,10 @@ lv_res_t lv_obj_del(lv_obj_t * obj)
         lv_ll_remove(&(par->child_ll), obj);
     }
 
+#if LV_USE_USER_DATA_FREE
+    /*Delete the user data*/
+    LV_USER_DATA_FREE(obj);
```

Comment:
How about we call it before/after `LV_EVENT_DELETE`? That way a user could also free the children's user data at the same time if they wanted to.

Reply:
I agree. Calling it after `lv_anim_del(obj, NULL);` would be better because there the user still has a fully working object and hence he has more freedom.

Reply:
Indeed, it may be more useful for functions to execute after `lv_anim_del(obj, NULL);`

Reply:
Great, could you update it?

Reply:
I'm not familiar with GitHub. So I resubmit the changes which 's easier for me. 
[new pull request](https://github.com/littlevgl/lvgl/pull/1386)

Reply:
No problem.
However, all you need to do is commit and push to your branch (from where the PR was created) and the commit will appear here.